api/api/clubs/tests.py

```diff
@@ -55,3 +58,133 @@ def testListClubs_hundredClubExistsInDb_returnsFiftyClubInFirstResponse(self):
+
+
+class UpdateClubsViewTest(APITestCase):
+    """Tests UpdateClubVew for updating club details """
+
+    @pytest.mark.django_db
+    def testUpdateClubs_nonAdminRequest_throwsForbidden(self):
+        # given
+        user = User.objects.create(
+            email='test@user.com', first_name='Test', last_name='User')
+        self.client.force_authenticate(user)
+
+        test_club = Club.objects.create(name="Bitbyte - The Programming Club",
+                                        category="S&T",
+                                        description="Some desc here",
+                                        email="theprogclub@iiitdmj.ac.in",
+                                        logo="#")
+        # when
+        response = self.client.put(f'/clubs/{test_club.id}/', {'name': 'Bitbyte',
+                                                               'category': "S&T",
+                                                               'description': "Some desc here",
+                                                               'email': "theprogclub@iiitdmj.ac.in",
+                                                               'logo': "https://iiitdmj.ac.in"})
+
+        # then
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    @pytest.mark.django_db
+    def testUpdateClubs_incompleteInformationInRequest_throwsBadRequest(self):
+        # given
+        user = User.objects.create(
+            email='test@user.com', first_name='Test', last_name='User')
+        self.client.force_authenticate(user)
+
+        test_club = Club.objects.create(name="Bitbyte - The Programming Club",
+                                        category="S&T",
+                                        description="Some desc here",
+                                        email="theprogclub@iiitdmj.ac.in",
+                                        logo="#")
+
+        Roles.objects.create(name='Coordinator',
+                             club=test_club,
+                             user=user,
+                             assigned_at=date.today())
+
+        # when
+        # missing club email
+        response = self.client.put(f'/clubs/{test_club.id}/', {'name': 'changed_club_name',
+                                                               'category': "S&T",
+                                                               'description': "Some desc here",
+                                                               'logo': "https://iiitdmj.ac.in"})
+
+        # then
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+
+# function based tests
+
+
+updated_club_data = [('name', 'changed_club_name'),
+                     ('category', 'Cultural'),
+                     ('description', 'some_desc_here'),
+                     ('email', 'changed@email.address'),
+                     ('logo', 'https://changedlogo.com')]
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('field, value', [
```

Comment:
I think this is not a good use of `parametrize`. For multiple input cases, you should use it.
Here the whole test would be run just to assert a different field value pair again.

You could simply assert each of these fields in the test.

Reply:
Is there a better way to use parametrize in this case then? 🤔

Reply:
Sorry misunderstood your last review. I'll make necessary changes.